x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentClusterServiceTests.java

```diff
@@ -2049,6 +2098,7 @@ private static DiscoveryNode buildNode(
     ) {
         return DiscoveryNodeUtils.builder(name)
             .name(name)
+            .ephemeralId(name)
```

Comment:
This doesn't reflect what happens in reality: https://github.com/elastic/elasticsearch/blob/27956c05c0bd2aae0037c2752b81db4b7dbf72a4/server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java#L405

So that means the new test isn't accurately reflecting what happens in real clusters either.

Reply:
OK, I see what's happening. With the new test the old and new cluster states are not being generated in a realistic way. Two completely different cluster states are generated with the same node names.

So for the _new_ test this change is OK. The node that is supposed to exist in both the new and old cluster states _hasn't_ been restarted in that test.

But my worry is that this change could mask a future bug that one of the existing tests would have found. I'll see if I can refactor things so that the new test passes without needing this change.